test/e2e/networking.go

```diff
@@ -282,3 +293,79 @@ func LaunchNetTestPodPerNode(f *Framework, nodes *api.NodeList, name, version st
 	}
 	return podNames
 }
+
+func getReadyNodes(f *Framework) (nodes *api.NodeList) {
+	nodes = ListSchedulableNodesOrDie(f.Client)
+	// previous tests may have cause failures of some nodes. Let's skip
+	// 'Not Ready' nodes, just in case (there is no need to fail the test).
+	filterNodes(nodes, func(node api.Node) bool {
+		return !node.Spec.Unschedulable && isNodeConditionSetAsExpected(&node, api.NodeReady, true)
+	})
+
+	if len(nodes.Items) == 0 {
+		Failf("No Ready nodes found.")
```

Comment:
Checking errors at each call point instead of only once here violates my sense of DRY, but I not have internalized the canonical way to do error handling in go.  I think in this case failing makes sense everywhere since no nodes means no test, but is it preferable to delay error handling as much as possible and I should be consistent with that philosophy?